libs/dealer.js

```
@@ -1,29 +1,106 @@
 
 const CARDS_PER_PLAYER = 2;
 
-class Dealer {
-    static dealCardToEachPlayer(players, deck, dealerPosition) {
-        for (let player of allPlayersFromDealer(players, dealerPosition)) {
-            player.cards.push(deck.drawCard());
-        }
+function clearActiveBets(table) {
+    table.players.forEach(player => { player.activeBetValue = -1; });
+    table.activeBetValue = 0;
+}
+
+function clearBoard(table) {
+    table.board = [];
+}
+
+function clearPot(table) {
+    table.potValue = 0;
+}
+
+function collectBlinds(table, players) {
+    table.activePlayerIndex = table.smallBlindPosition;
+    table.activePlayer = players[table.activePlayerIndex];
+    handleRaise(5, table);
+    table.activePlayerIndex = table.bigBlindPosition;
+    table.activePlayer = players[table.activePlayerIndex];
+    handleRaise(5, table);
+}
+
+function dealCardToEachPlayer(players, deck, dealerPosition) {
+    for (let player of allPlayersFromDealer(players, dealerPosition)) {
+        player.cards.push(deck.drawCard());
     }
-    
-    static dealNewCardsToEachPlayer(players, deck, dealerPosition) {
-        players.forEach(player => { player.cards = []; });
-        for (let i = 0; i < CARDS_PER_PLAYER; i++) {
-            this.dealCardToEachPlayer(players, deck, dealerPosition);
-        }
+}
+
+function dealNewCardsToEachPlayer(table, players, dealerPosition) {
+    table.deck = new Deck();
+    table.deck.shuffle();
+    players.forEach(player => { player.cards = []; });
+    for (let i = 0; i < CARDS_PER_PLAYER; i++) {
+        dealCardToEachPlayer(players, table.deck, dealerPosition);
     }
+}
 
-    static dealStreet(board, deck) {
-        deck.drawCard(); // Burn
-        if (!board.length) { // Flop
-            board.push(deck.drawCard());
-            board.push(deck.drawCard());
-        }
-        board.push(deck.drawCard()); // Turn & River
+function dealStreet(board, deck) {
+    deck.drawCard(); // Burn
+    if (!board.length) { // Flop
```

You could throw an error if board is over length 5